### ISSUE TITLE
Improve scaling of compass fields on low res devices

### ISSUE THREAD

**Issue by Lineflyer** [cgeo/cgeo#9051]
Taken from FB:

Text:
How do I get rid of these 2 banners as they are blocking out the compass..this has happened on the latest beta version.

![grafik](https://user-images.githubusercontent.com/949669/94058595-83dd2a80-fde1-11ea-8b2d-28a999981bf4.png)

**Comment by eddiemuc** [comment]
Looks bad indeed. Was not aware that such small screens are still used these days. Alas, only the second "banner" is new (and the scrambled-up text directly underneath), the first one was there before.

I have no real good idea on how to deal with this with regards to GUI-design. Does anyone have an idea? Maybe a toggle button "Compact view/normal view" in the context menu (showing and hiding the two banners as well as the text line underneath them)? Also, one would have to add the functionality of the two banners into the context menu then additionally.

**Comment by geocachermgo** [comment]
My ideas (to be filtered)
reduce the text eg from "GPS and magnetic compass" to "GPS and compass", together with "Flat" mode this would save 2 lines and compass would be fully visible. 
- Reduce the font of bearing and distance
- Only one bar with toogle: GPS only, auto, flat, upright (not sure if I would like this, but advantage would be the appearing / disappearing of the second bar would not be necessary)
For the scrambeld text:
-omit orientation + values in case of low resolution
-exchange text "device heading" and number. Then there would be scambling in the middle but the numbers would be readable.

**Comment by geocachermgo** [comment]
Not the compass topic: Would be intesting how a list with long names like GCCC88 would look on that device after #8966 ?

**Comment by rsudev** [comment]
Do you now the specs from this device? The resolution seems to be FullHD (if this is a real screenshot). Would be interesting to know the dpi value.

**Comment by Lineflyer** [comment]
I will try to find out.

**Comment by geocachermgo** [comment]
With split screen you can also get small screens:
![Screenshot_2020-09-24-21-36-03-398_cgeo geocaching](https://user-images.githubusercontent.com/43813636/94191623-10551f00-feae-11ea-90c5-43e71eaaeb63.jpg)

**Comment by geocachermgo** [comment]
![IMG_20200924_214437](https://user-images.githubusercontent.com/43813636/94192622-7db57f80-feaf-11ea-91c5-fdcb6254fba0.jpg)

**Comment by geocachermgo** [comment]
> I will try to find out.

OnePlus 5 with Android 10
14 cm (5,5 Zoll), 1920 × 1080 PenTile-Pixel (401 ppi)

**Comment by geocachermgo** [comment]
If you choose extrem android settings of font and symbol size the compass layout gets also ugly.

**Comment by eddiemuc** [comment]
Is there some sort of "official" statement up to which screen sizes we support proper layout?

Since this is imho a corner case and also a release problem (thus no big code changes should be introduced) I propose a simple solution:
* Shorten the text "GPS and magnetic compass used" to "All Sensors used" / "Alle Sensoren" (which is closer to the truth anyway)
* Shorten the text "Device orientation: upright (Auto)" to "Device is: upright (Auto)" / "Gerät ist: aufrecht (Auto)"
* Shorten small labels (in text line under second button) "Device heading" to "Device"/"Gerät" (trusting that users get this is the heading from the context - it is directly placed over the cache heading where also no label seems to be necessary)

Ok?

**Comment by geocachermgo** [comment]
On FB the user confirmed that she uses a big font due to poor eye sight problems.

**Comment by geocachermgo** [comment]
> 
> Ok?

Ok, from my side

**Comment by Lineflyer** [comment]
> On FB the user confirmed that she uses a big font due to poor eye sight problems

Ok, so its a special case, which is hard to cover.

> Shorten the text "GPS and magnetic compass used" to "All Sensors used" / "Alle Sensoren" (which is closer to the truth anyway)

Ok, english could also be "All sensors" without "used"

> Shorten the text "Device orientation: upright (Auto)" to "Device is: upright (Auto)" / "Gerät ist: aufrecht (Auto)"

Could better be "Orientation: Upright (Auto)"

> Shorten small labels (in text line under second button) "Device heading" to "Device"/"Gerät" (trusting that users get this is the heading from the context - it is directly placed over the cache heading where also no label seems to be necessary)

IMHO we can remove them completely or only show them if "Debug logging" is active. We used the debug log switch in the past already to also show more on screen info (e.g. on live map title).

**Comment by geocachermgo** [comment]
I would agree for the right "orientation" section with azimuth, pitch which I see als a (really) nice to have but would strongly oppose regarding device heading. That display works (for me) much better than the compass rose.
The user did not complain about the scrambeled small info but only about the clipped compass.

Since there is option for additional info:
Shall we have a general "simple" option with reduced menues, display, etc ("Seniorentelefon")?

**Comment by rsudev** [comment]
From what I understand the title of this issue should rather be: "Improve compass UI on devices with large font settings" - I would not put too much effort in such an edge case, or even worse, degrade the experience for the 99% on the way.
As far as I can tell it looks ugly in this case, but is usable.

**Comment by eddiemuc** [comment]
I am willing to offer a PR changing labels as suggested and ensuring that button text does not spill over one line.
I can't offer a PR for removing the device heading and azi/pitch/roll information or reduce their visibility - I put too much love into these...
Such a PR has to go on release branch, right (because it is from beta test)

**Comment by Lineflyer** [comment]
> I am willing to offer a PR changing labels as suggested and ensuring that button text does not spill over one line.
> I can't offer a PR for removing the device heading and azi/pitch/roll information or reduce their visibility - I put too much love into these...
> Such a PR has to go on release branch, right (because it is from beta test)

Sounds a good compromise. IMHO the shorter labels, etc. should be enough. Its a special case with a low res device combined with enlarged fonts.

The PR should target `release`. We want to me branches anyway, but who knows what might come. So `release` is better.

**Comment by eddiemuc** [comment]
Just wanted to fix this but I saw that the new button/functionality is NOT on release branch :-)
So I will fix on master, but I wonder why user reported this in beta - I thought beta was taken from release branch?

**Comment by geocachermgo** [comment]
> So I will fix on master, but I wonder why user reportedthis in beta - I thought beta was taken from release branch?

She was using a nightly.

**Comment by Lineflyer** [comment]
The user confused beta with nightly.
